code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm

```diff
@@ -34,6 +34,16 @@
 /datum/ai_behavior/spiderling/New(loc, parent_to_assign, escorted_atom, can_heal = FALSE)
 	. = ..()
 	RegisterSignal(escorted_atom, COMSIG_XENOMORPH_ATTACK_LIVING, .proc/go_to_target)
+	RegisterSignal(escorted_atom, COMSIG_XENOMORPH_ATTACK_OBJ, .proc/go_to_obj_target)
+
+/// Signal handler to check if we can attack the obj's that our escorted_atom is attacking
+/datum/ai_behavior/spiderling/proc/go_to_obj_target(source, obj/target)
+	SIGNAL_HANDLER
+	if(isnull(target))
+		return
+	if(target.obj_integrity <= 0)
```

Comment:
Why are you checking that

Reply:
To avoid the obj taking damage past being QDELeted, got some local runtimes when too many spiderlings attacked